src/components/Link.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+
+type LinkVariantProps = React.HTMLProps<HTMLAnchorElement> & {
+  isActive: boolean;
+  variant?: "link";
+  href: string;
+  setIsActive: (newValue: boolean) => void;
+};
+type NavVariantProps = React.HTMLProps<HTMLLIElement> & {
+  isActive: boolean;
+  variant: "nav";
+  onClick: () => void;
+};
+type Props = LinkVariantProps | NavVariantProps;
+
+/**
+ * A reusable link component with animated underline when the link is active.
+ *
+ * Has two variants:
+ *  1. "link" (default): renders an <a> tag and calls a setIsActive callback
+ *     when the link is (un)focused and (un)hovered.
+ *  2. "nav": renders a <li> tag with an onClick callback. Has no inherent
+ *     hover/focus behaviour.
+ */
+const Link = (props: Props) => {
+  const { isActive, variant, children } = props;
+
+  const className = `
+    ${variant === "nav" ? "mx-4" : null}
```

Comment:
The `mx-4` should probably more on an element such as the `li` itself, rather than in `Link`.